lhotse/cut/multi.py

```diff
@@ -0,0 +1,1128 @@
+import logging
+from dataclasses import dataclass, field
+from functools import partial
+from math import isclose
+from typing import Any, Callable, Dict, Iterable, List, Optional, Union
+
+import numpy as np
+import torch
+from intervaltree import IntervalTree
+
+from lhotse.audio import Recording
+from lhotse.augmentation import AugmentFn
+from lhotse.cut.base import Cut
+from lhotse.features import FeatureExtractor, Features
+from lhotse.features.io import FeaturesWriter
+from lhotse.supervision import SupervisionSegment
+from lhotse.utils import (
+    LOG_EPSILON,
+    Pathlike,
+    Seconds,
+    TimeSpan,
+    add_durations,
+    compute_num_frames,
+    compute_num_samples,
+    fastcopy,
+    ifnone,
+    measure_overlap,
+    overlaps,
+    overspans,
+    perturb_num_samples,
+    rich_exception_info,
+    uuid4,
+)
+
+
+@dataclass
+class MultiCut(Cut):
+    """
+    :class:`~lhotse.cut.MultiCut` is a :class:`~lhotse.cut.Cut` that is analogous to the MonoCut.
+    While MonoCut represents a single channel of a recording, MultiCut represents multiple channels
+    that share the same supervision. It is intended to be used to store, for example, segments
```

Comment:
Just skimming through for now, but are you sure you want to the supervisions to always be shared across channels? Can we envision a MultiCut that has different supervisions on each channel?